velox/type/DecimalUtil.h

```diff
@@ -147,6 +148,43 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleFloating(
+      const TInput inputValue,
+      const int toPrecision,
+      const int toScale) {
+    auto toValue =
+        inputValue * static_cast<TInput>(DecimalUtil::kPowersOfTen[toScale]);
+
+    bool isOverflow = !std::isfinite(toValue);
```

Comment:
If `toValue` is not infinitty, `isOverflow` is set as true. Is this correct?

Reply:
The function `std::isfinite` will return false only if the `toValue` is `NAN` or `INFINITY`. So if `toValue` is not infinity or nan, `isOverflow` will be set as false.

Reply:
Sorry, a missing reading to `std::isinf`.